src/client/Client.cc

```diff
@@ -9243,14 +9244,16 @@ void Client::seekdir(dir_result_t *dirp, loff_t offset)
 //};
 void Client::fill_dirent(struct dirent *de, const char *name, int type, uint64_t ino, loff_t next_off)
 {
-  strncpy(de->d_name, name, 255);
-  de->d_name[255] = '\0';
+  size_t len = strlen(name);
+  len = std::min(len, (size_t)255);
+  memcpy(de->d_name, name, len);
+  de->d_name[len] = '\0';
```

Comment:
I think snprintf() adds unnecessary overhead for a simple string copy. It can also return errors which would need to be checked (man page doesn't explicitly says what errors can happen, so I think we can't assume it will never fail). Additionally, if the string is truncated, the returned value is not the current length, but the needed size for the buffer, so additional checks are needed.

I don't see it as a better alternative.

Reply:
> I think snprintf() adds unnecessary overhead for a simple string copy. It can also return errors which would need to be checked (man page doesn't explicitly says what errors can happen, so I think we can't assume it will never fail). 

Based on usage you can say it cannot fail.

> Additionally, if the string is truncated, the returned value is not the current length, but the needed size for the buffer, so additional checks are needed.

That's a feature!

> I don't see it as a better alternative.

Okay.